### ISSUE TITLE
Support for Pandoc-flavored Markdown span links

### ISSUE THREAD

**Issue by paanvaannd** [jakewvincent/mkdnflow.nvim#64]
Thanks for the great plugin!

I'm looking to anchor link to arbitrary spans of text within Markdown both within and between Markdown files. This is possible in Pandoc-flavored Markdown through the use of [spans](https://pandoc.org/MANUAL.html#divs-and-spans). Would it be possible to integrate span links through mkdnflow?

Concrete case: file `a.md` has text `"hello [world](b.md#planet-earth)"` that, when followed, would lead to `b.md` and place the cursor on "P" (or anywhere in the span) within the sentence `"[Planet Earth]{#planet-earth} is the third planet from the Sun."` Another simple example can be seen in [this related Stack Overflow answer](https://stackoverflow.com/a/38419698).

Basically, it would follow the same behavior as following anchor links to headers within the same/other documents but with spans instead.

**Comment by jakewvincent** [comment]
@paanvaannd feedback is welcome on the above two situations, btw! For your use case, what would your preferences be in those situations?

**Comment by paanvaannd** [comment]
Great points!

I like your proposed idea for auto-searching. Perhaps the default behavior can generalize to both cases: just land on either the first or last instance of a target ID and potentially trigger an auto-search whether the instance is a heading or a span.

What's your preference for how opinionated `mkdnflow` behavior should be? From the documentation, I infer it leans towards opinionated. For example, the default behavior for handling non-existent directories infers do-what-I-mean behavior for auto-creating them rather than prompting the user for a choice.

If you want a default behavior (perhaps even non-configurable), my two cents for both cases would be that, per [The Zen of Python](https://peps.python.org/pep-0020/), "Explicit is better than implicit." The user went out of their way to manually ID a span with some text, so favor that explicit declaration over inferred meaning from headers. For case \#2, they explicitly named at least two spans with the same ID, so infer that they favor the last span with that ID than any other. If they want differentiation, they should explicitly differentiate. Perhaps it would even be best to follow the canonical behavior of Pandoc in this matter, but I'm not sure (and can't find off of cursory searches) what Pandoc's default behavior for resolving ambiguity is.
 
If you are open to letting the user specify behavior, then I would say config options are better than prompts as the latter would likely be intrusive. Perhaps the decision could even even be split into two options:

1. `favor_first_ambiguous_target` (`bool`): follow the first ('true') or last ('false') instance of a duplicate target ID, whether they are headers or spans; default to `true` (for e.g.)
2. `trigger_auto_search` (`bool`): execute an auto-search for other instances that users can `n`/`N` through upon navigation to ambiguous target; default to `true` (for e.g.)

Additionally, if there is a default or non-configurable behavior, it should be documented\* and/or explicitly reported to the user. I think ambiguity in a case that should be unique is an error, so throwing an error would be warranted. Per the "Errors should never pass silently \ Unless explicitly silenced" principles , perhaps there could be an error/warning thrown with output from `vim.notify()"` declaring the executed behavior and issue (e.g., `WARN: ambiguous reference, linking to last instance of ambiguous target`). Finally, that presents opportunity for a third potential config option such as 'warn_on_ambiguous_target` (`bool` defaulting to `true`) so that if the user if happy with either the default (configurable or not) or configured action, they can just be left alone.

\* as you already do: one of the most clearly-documented programs I've seen!

**Comment by jakewvincent** [comment]
I appreciate the compliment! I *want* to make the documentation better--I always update the readme, but I let the help docs definitely lag a little too far behind the updates.

I just added an initial attempt at this feature to the dev branch. It should allow linking to explicitly tagged spans in other documents (e.g. `[Link](file#tagged-span)` with implicit extensions enabled or `[Link](file.md#tagged-span)` without) as well as explicitly tagged spans in the current document (e.g. `[Link](#tagged-span)`). I liked your suggestion to prioritize explicit over implicit, so what the link-following function will now do is first scan the file in question for an explicit span, and if none is found, then it will look for a heading. I haven't added any ambiguity resolution yet, and perhaps I'll just leave it as is--stopping at the first match. This seems reasonable enough since it's a little more efficient and since IDs are generally expected to be unique.

When you get a chance, would you mind pulling the dev branch and testing this out?

```lua
use({
    'jakewvincent/mkdnflow.nvim',
    branch = 'dev',
    config = function()
        require('mkdnflow').setup({
            -- leave blank for defaults
        })
    end
})
```

**Comment by paanvaannd** [comment]
Thank you for the updates and apologies for the late response! The past two weeks have been terribly busy, but I may have time later this week to test out the implementation and offer more feedback. I'll update you as soon as I'm able to test it out!

**Comment by paanvaannd** [comment]
I was finally able to test the implementation and I love it: thank you so much!

One thing I noticed is that links don't seem to find bracketed spans that precede their linked reference within the same file.

E.g., the following will work within the same file:

```
I learned a [strange incantation](#cthulhu-incantation) today!

...

> [Ph'nguli mglw'nafh Cthulu R'lyeh wgah'nagl fhtang]{#cthulhu-incantation}
```

but not this:

```
> [Ph'nguli mglw'nafh Cthulu R'lyeh wgah'nagl fhtang]{#cthulhu-incantation}

...

I learned a [strange incantation](#cthulhu-incantation) today!
```

Cross-document linking works perfectly for my tested cases, though!

**Comment by jakewvincent** [comment]
Thanks for testing it! I tested out following links to preceding bracketed spans and can't get it to jump to those either. Will fix that.

**Comment by jakewvincent** [comment]
Should be fixed now! (commit 962c9fba74fa4aec611280d5d6da10ae0a564dd5)

**Comment by paanvaannd** [comment]
That was quick! Thanks for your hard work on this: it will really help me with organizing my life and work.

I'd love to donate to help support your work as well but I can't find an Open Collective, GitHub Sponsors, Patreon, Buy Me a Coffee, etc. link anywhere! Or perhaps a donation to a charity/cause of your choice?

**Comment by jakewvincent** [comment]
ahh, I really appreciate the desire to contribute! I already enjoy just working on the plugin as a hobby, but I won't turn down contributions. I just added a ko-fi donation link to the readme. Thanks so much!
